library_generation/utilities.sh

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env bash
+
+set -xeo pipefail
+
+# define utility functions
```

Comment:
Can we have "unit tests" for functions in this file?

Reply:
Can I add unit tests in following PRs?
I have added an integration test in this PR.

Reply:
Sure, you can do it in a separate PR.